### ISSUE TITLE
[Vote] Should we encourage coding with unless?

### ISSUE THREAD

**Issue by kbrock** [ManageIQ/guides#428]
See https://www.rubydoc.info/github/bbatsov/RuboCop/RuboCop/Cop/Style/NegatedIf

For negative conditions, We currently discourage the use of `if !condition` and ask the developer to replace it with `unless condition`
This is true with prefix and postfix conditions.

```
unless foo # prefix using unless
  bar
end

if !foo #prefix using if
  bar
end

bar unless foo # postfix using unless
bar if !foo # postfix using if
```

The consistency is sometimes helpful, but at other times it puts unwanted emphasis on the negative.

Do you want to encourage developers to use `unless`?

- :+1: Encourage developers to have consistent code. Say no to bike shedding. (`keep` the cop)
- :-1: Let developers express the emphasis they want. (`disable` the cop)

If we do encourage consistency, how do you like to see it? (vote regardless of how you voted above)

- :heart: I love them on their own line. It make it more readable (`prefix` +1)
- :confused: Those `unless` statements on their own line confuse me (`prefix` -1)
- :rocket: Having them at the end of the line is out of this world. (`postfix` +1)
- :eyes: Having them on the end of the line burns my eyes. (`postfix` -1)

Of note, the sheer quantity of parameters here states that there are a slew of preferences in this subject.

**Comment by carbonin** [comment]
Even after years, I still convert `unless` to "if not ..." in my head every time.

**Comment by NickLaMuro** [comment]
Honestly, I think this cop is confusing to start with... seems like it is trying to combine two separate concerns into one cop.

Surprised there isn't a "Don't use `unless`" cop, because it seems like that would be the middle ground for the 👍 's and the 👎 's in this poll.

I think getting rid of the cop is the simplest solution, and we can self regulate the parts this cop attempts to do.

* If you are using a `postfix`, avoid multiple conditionals (or convert to a method?)
* If using a prefix, and it is just one conditional, consider a `postfix` if it is more readable
* If you are using `unless`, avoid multiple conditionals

Things like that.

**Comment by kbrock** [comment]
there is a cop to encourage postfix (regardless of `if` or `unless`. Not sure about discouraging it.

Just writing this up made me feel like this is 2 or 3 cops

**Comment by agrare** [comment]
> I think getting rid of the cop is the simplest solution, and we can self regulate the parts this cop attempts to do.

Yeah I completely agree with this, I would like to not be discouraged from using `if !something` (aka disable the cop) and if you do use unless it has to be super simple like `unless foo.nil?` (I don't care too much about prefix or postfix, mostly just if you use unless its got to be a single condition)

> Even after years, I still convert unless to "if not ..." in my head every time.

This, so much this
